csrc/device_lower/pass/scalar_hoist.cpp

```diff
@@ -342,11 +342,19 @@ std::vector<Val*> getAssumptions(const std::vector<kir::ForLoop*>& loops) {
     Val* start = loop->start();
     assumptions.push_back(IrBuilder::geExpr(loop->index(), start));
     Val* stop = loop->simplifiedStop();
-    if (!stop->sameAs(start)) {
+    if (stop->sameAs(start)) {
       // If stop = start, then this loop will not be computed, so it's not
       // important to simplify its index. However, it is important that we avoid
       // contradicting assumptions, so we omit the index < stop assumption in
       // these cases.
+      TORCH_WARN(
```

Comment:
No need to be this PR: should we add a `NVF_WARN`?

Reply:
We definitely should, if for no other reason than to capture the output in python using debug().